cmd/booster-http/server.go

```diff
@@ -302,27 +305,45 @@ func serveContent(w http.ResponseWriter, r *http.Request, content io.ReadSeeker,
 	// try to do it implicitly
 	w.Header().Set("Content-Type", contentType)
 
-	if r.Method == "HEAD" {
-		// For an HTTP HEAD request ServeContent doesn't send any data (just headers)
-		http.ServeContent(w, r, "", time.Time{}, content)
-		alog("%s\tHEAD %s", color.New(color.FgGreen).Sprintf("%d", http.StatusOK), r.URL)
-		return
-	}
-
-	// Send the CAR file
 	// http.ServeContent ignores errors when writing to the stream, so we
 	// replace the writer with a class that watches for errors
 	var err error
 	writeErrWatcher := &writeErrorWatcher{ResponseWriter: w, onError: func(e error) {
 		err = e
 	}}
 
-	// Send the content
 	// Note that the last modified time is a constant value because the data
 	// in a piece identified by a cid will never change.
 	start := time.Now()
 	alogAt(start, "%s\tGET %s", color.New(color.FgGreen).Sprintf("%d", http.StatusOK), r.URL)
-	http.ServeContent(writeErrWatcher, r, "", lastModified, content)
+	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+		// If Accept-Encoding header contains gzip then send a gzipped response
+
+		gzipwriter := &gziphandler.GzipResponseWriter{
+			ResponseWriter: writeErrWatcher,
+		}
+		// Set the Content-Encoding header to gzip
+		w.Header().Set("Content-Encoding", "gzip")
+		if r.Method == "HEAD" {
+			// For an HTTP HEAD request ServeContent doesn't send any data (just headers)
+			http.ServeContent(w, r, "", time.Time{}, content)
+			alog("%s\tHEAD %s", color.New(color.FgGreen).Sprintf("%d", http.StatusOK), r.URL)
+			return
+		}
+
+		// Send the content
+		http.ServeContent(gzipwriter, r, "", lastModified, content)
```

Comment:
This block of code is the same in both parts of the if / else statement.
I'd suggest instead doing something like:
```
var content http.ResponseWriter
if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
    ...
    content = gziphandler....
}
if r.Method == "HEAD" {
    // For an HTTP HEAD request ServeContent doesn't send any data (just headers)
    http.ServeContent(w, r, "", time.Time{}, content)
    alog("%s\tHEAD %s", color.New(color.FgGreen).Sprintf("%d", http.StatusOK), r.URL)
    return
}

// Send the content
http.ServeContent(content, r, "", lastModified, content)
```